packages/n4s/src/enforce/compounds/oneOf.js

```diff
@@ -0,0 +1,9 @@
+import runLazyRules from 'runLazyRules';
+
+/**
+ * @param {*} value   Value to be test against rules
+ * @param {Function[]} rules    Rules to validate the value with
+ */
+export default function oneOf(value, ...rules) {
+  return (rules) && (rules.filter(ruleGroup => runLazyRules(ruleGroup, value)).length === 1);
+}
```

Comment:
Not a problem with your code - but I discovered recently that transpiling to ES5 actually makes rest parameters actually pretty large in terms of bundle size.

I added a simple workaround that mimics this behavior, but reduces its size quite dramatically

What the following change does is basically wrapping the function with a higher order function that passes down all the arguments as `first` argument, and the rest of the arguments as an array.

```suggestion
import runLazyRules from 'runLazyRules';
import { withFirst } from 'withArgs';

/**
 * @param {*} value   Value to be test against rules
 * @param {Function[]} rules    Rules to validate the value with
 */
function oneOf(value, rules) {
  return (rules) && (rules.filter(ruleGroup => runLazyRules(ruleGroup, value)).length === 1);
}

export default withFirst(oneOf)
```